torch/testing/_internal/custom_op_db.py

```diff
@@ -0,0 +1,112 @@
+import torch
+from torch.testing._internal.opinfo.core import (
+    OpInfo,
+)
+from torch.testing._internal.common_dtype import all_types_and
+import numpy as np
+from torch._custom_op import CustomOp
+from torch.testing._internal.autograd_function_db import (
+    sample_inputs_numpy_cube,
+    sample_inputs_numpy_mul,
+    sample_inputs_numpy_sort,
+    sample_inputs_numpy_take,
+)
+
+# Note: [custom op db]
+#
+# This is a collection of custom operator test cases written as OpInfos
+# so they can easily be consumed by OpInfo-based tests to check if subsystems
+# support them correctly.
+
+def to_numpy(tensor):
+    return tensor.cpu().numpy()
+
+numpy_cube = CustomOp.define('_torch_testing::numpy_cube(Tensor x) -> (Tensor, Tensor)')
+
+@numpy_cube.impl('cpu')
+@numpy_cube.impl('cuda')
+def numpy_cube_impl(x):
+    x_np = to_numpy(x)
+    dx = torch.tensor(3 * x_np ** 2, device=x.device)
+    return torch.tensor(x_np ** 3, device=x.device), dx
+
+@numpy_cube.impl_meta()
+def numpy_cube_meta(x):
+    return x.clone(), x.clone()
+
+numpy_mul = CustomOp.define('_torch_testing::numpy_mul(Tensor x, Tensor y) -> Tensor')
+
+@numpy_mul.impl('cpu')
+@numpy_mul.impl('cuda')
+def numpy_mul_impl(x, y):
+    return torch.tensor(to_numpy(x) * to_numpy(y), device=x.device)
+
+@numpy_mul.impl_meta()
+def numpy_mul_meta(x, y):
+    return x * y
```

Comment:
This meta is not actually right 😛

Reply:
Yeah, you're going to get the stock multiply meta behavior which is too flexible